pkg/nat/nat.go

```diff
@@ -117,23 +91,20 @@ func CreateNAT(
 
 		namespaceVeths     []*network.IP
 		namespaceVethsLock sync.Mutex
-
-		claimableNamespaces     = map[string]claimableNamespace{}
-		claimableNamespacesLock sync.Mutex
 	)
 
 	var closeLock sync.Mutex
 	closed := false
 
-	closeInProgressContext, cancelCloseInProgressContext := context.WithCancel(closeCtx) // We use `closeContext` here since this simply intercepts `ctx`
+	closeInProgressContext, cancelCloseInProgressContext := context.WithCancel(rescueCtx) // We use `closeContext` here since this simply intercepts `ctx`
```

Comment:
```suggestion
	closeInProgressContext, cancelCloseInProgressContext := context.WithCancel(rescueCtx) // We use `rescutContext` here since this simply intercepts `ctx`
```

Reply:
Resolved in 480291b9f632c2f188f6f4a086beba8efdbf2c00